data/dataset.py

```diff
@@ -0,0 +1,129 @@
+
+        # change tempo augmentation
+        if random.random() < self.time_stretch_probability:
+            record["dstart"], record["duration"] = change_speed(dstart=record["dstart"], duration=record["duration"])
+
+        return record
+
+    def apply_masking(self, token_ids: np.ndarray, record: dict):
+        input_token_ids = token_ids.copy()
+        tgt_token_ids = token_ids.copy()
+
+        # masking, adds new key to record dict called masked
+        masked, mask_type = self.masks.apply(record, p=self.masking_probability)
+
+        # source token ids
+        mask_idx = self.tokenizer.token_to_id["<mask>"]
+        input_token_ids[masked] = mask_idx
+
+        # tgt token ids, -100 means loss is not calculated on this token
+        tgt_token_ids[~masked] = -100
+
+        # add mask type token
+        mask_type_idx = self.tokenizer.token_to_id[mask_type]
+        input_token_ids = np.insert(input_token_ids, obj=0, values=mask_type_idx)
+        tgt_token_ids = np.insert(tgt_token_ids, obj=0, values=-100)
+
+        return input_token_ids, tgt_token_ids
+
+    def add_cls_token(self, input_token_ids: np.ndarray, tgt_token_ids: np.ndarray):
+        cls_token = self.tokenizer.token_to_id["<cls>"]
+        input_token_ids = np.insert(input_token_ids, obj=0, values=cls_token)
+        tgt_token_ids = np.insert(tgt_token_ids, obj=0, values=-100)
+
+        return input_token_ids, tgt_token_ids
+
+    def __getitem__(self, index: int) -> dict:
+        record = self.dataset[index]
+
+        filename = record["midi_filename"]
+
+        # sanity check, replace NaN with 0
+        if np.any(np.isnan(record["dstart"])):
+            record["dstart"] = np.nan_to_num(record["dstart"], copy=False)
+
+        record = self.apply_augmentation(record)
+        record = self.quantizer.quantize_record(record)
+        token_ids = self.tokenizer.encode(record)
+        input_token_ids, tgt_token_ids = self.apply_masking(token_ids, record)
+        input_token_ids, tgt_token_ids = self.add_cls_token(input_token_ids, tgt_token_ids)
+
+        tokens = {
+            "filename": filename,
+            "source_token_ids": torch.tensor(token_ids, dtype=torch.long),
+            "input_token_ids": torch.tensor(input_token_ids, dtype=torch.long),
+            "tgt_token_ids": torch.tensor(tgt_token_ids, dtype=torch.long),
+        }
+
+        return tokens
+
+
+if __name__ == "__main__":
+    from omegaconf import DictConfig
+    from datasets import load_dataset
+    from torch.utils.data import DataLoader
+
+    from data.tokenizer import QuantizedMidiEncoder
+
+    quantization_cfg = DictConfig(
+        {
+            "dstart": 7,
+            "duration": 7,
+            "velocity": 7,
+        }
+    )
+
+    ds = load_dataset("JasiekKaczmarczyk/maestro-v1-sustain-masked", split="train")
+
+    quantizer = MidiQuantizer(7, 7, 7)
+    tokenizer = QuantizedMidiEncoder(7, 7, 7)
```

Comment:
```suggestion
    ds = load_dataset("JasiekKaczmarczyk/maestro-v1-sustain-masked", split="train")
    quantization_cfg = DictConfig(
        {
            "dstart": 7,
            "duration": 7,
            "velocity": 7,
        }
    )
    quantizer = MidiQuantizer(quantization_cfg=quantization_cfg)
    tokenizer = QuantizedMidiEncoder(quantization_cfg=quantization_cfg)
```

I know this is a testing script, but it should demonstrate how to use our elements in an elegant way. If you don't want to update the quantizer and encoder classes, you can also do:

```python
quantizer = MidiQuantizer(
    dstart_bins=quantizer_cfg.dstart,
    duration_bins=quantizer_cfg.duration,
    velocity_bins=quantizer_cfg.velocity,
)
```